### PR TITLE
OSLC Configuration Context support

### PR DESCRIPTION
**Pull request by cenobyte321**
An approach to add support for Configuration Context in OSLC4J Client requests and OSLC queries.

### CODE REVIEW

**Comment by berezovskyi** [comment]
Thanks Luis, I will take a look at it today! 👏

File: oslc4j-client/src/main/java/org/eclipse/lyo/oslc4j/client/resources/RmUtil.java [hunk 1]
```diff
@@ -56,7 +59,12 @@ public static ResourceShape lookupRequirementsInstanceShapes(final String servic
 									URI[] instanceShapes = creationFactory.getResourceShapes();
 									if (instanceShapes != null ){
 										for ( URI typeURI : instanceShapes) {
-											response = client.getResource(typeURI.toString(),OSLCConstants.CT_RDF);
+											if(configurationContext != null) {
+												response = client.getResource(typeURI.toString(), null, OSLCConstants.CT_RDF, configurationContext);
+											}
+											else{
```
Comment (berezovskyi) [inline oslc4j-client/src/main/java/org/eclipse/lyo/oslc4j/client/resources/RmUtil.java]: Do we need this? I thought your code handles null as a missing context.

Comment (cenobyte321) [reply oslc4j-client/src/main/java/org/eclipse/lyo/oslc4j/client/resources/RmUtil.java]: You're right. This is a redundant check.

File: oslc4j-client/src/main/java/org/eclipse/lyo/oslc4j/client/OslcClient.java [hunk 2]
```diff
@@ -63,6 +63,7 @@
  */
 public class OslcClient {
 
+	public static final String CONFIGURATION_CONTEXT_HEADER = "Configuration-Context";
```
[the remark this answers is not in the document]
Comment (cenobyte321) [reply oslc4j-client/src/main/java/org/eclipse/lyo/oslc4j/client/OslcClient.java]: Understood, I'll move it there.

**Comment by berezovskyi** [comment]
Looks good, Luis! We can merge this as soon as you sign the ECA. You will need to register an Eclipse account for that. Thanks a lot for the contribution!

**Comment by cenobyte321** [comment]
> Looks good, Luis! We can merge this as soon as you sign the ECA. You will need to register an Eclipse account for that. Thanks a lot for the contribution!

My pleasure! I just signed the ECA and signed the commits of this pull request.

**Comment by berezovskyi** [comment]
@cenobyte321 thanks for your first contribution to Lyo! Merged to master.

**Pull request merged**